### PR TITLE
Hoist Astro.globbed hoisted scripts in dev

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- This uses the Vite graph to discover scripts that need to be hoisted, similar to how we do so for styles.
- Previously was using the Astro metadata `hoistedScriptPaths()` method, but this does not traverse through Astro.globs, or through JS re-export pattern. Using the Vite graph ensures we get all hoisted scripts.
- This is how it's done in the build as well.
- Bug reported on Discord: https://discord.com/channels/830184174198718474/996921397978857522

## Testing

- Test added

## Docs

N/A, bug fix

### CODE REVIEW

**Comment by matthewp** [comment]
Holding off on merging since this is a biggish change to dev.

**Review by natemoo-re — APPROVED** [review]
LGTM once we're ready to merge this next week!

**Pull request merged**
